swarm/api/content_type.go

```diff
@@ -0,0 +1,57 @@
+package api
+
+import (
+	"fmt"
+	"mime"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// detect content type by file content
+// otherwise by file extension
+// returns "application/octet-stream" in worst case
+func DetectContentType(file string) string {
+	var contentType = "application/octet-stream" // default value
+
+	f, err := os.Open(file)
+	if err != nil {
+		log.Warn("detectMimeType: can't open file", "file", file, "err", err)
```

Comment:
`log.Error` - err != nil
also - you're not returning on error (the rest of the code below will continue executing - resulting in an error). if you choose to have another handle opened here - please return an empty string in this code block - the function has to return here.
as mentioned above - i would remove this part and already use the open handle that is created in the enclosing call